router/router_spec.go

```diff
@@ -0,0 +1,12 @@
+package router
+
+type PutVendorSpec struct {
```

Comment:
don't you think it's better if we move this to `/vendor` package? specifically `vendor.go`